source/GameInterface/Serialization/Native/KeyValuePairBinaryPackage.cs

```diff
@@ -60,5 +63,27 @@ public T Unpack<T>(IBinaryPackageFactory binaryPackageFactory)
         {
             return (T)Unpack(binaryPackageFactory);
         }
+        
+        private void UnpackInternal()
+        {
+            var type = Object.GetType();
+            var fields = type.GetAllInstanceFields();
+
+            foreach (string fieldName in StoredFields.Keys)
+            {
+                var field = fields.FirstOrDefault(f => f.Name.Equals(fieldName));
+
+                if (type.IsValueType)
+                {
+                    object boxed = Object;
+                    field.SetValue(boxed, StoredFields[fieldName].Unpack(binaryPackageFactory));
+                    Object = boxed;
+                }
+                else
+                {
+                    field.SetValue(Object, StoredFields[fieldName].Unpack(binaryPackageFactory));
+                }
```

Comment:
Would this do the same thing?
```suggestion
                field.SetValue((object)Object, StoredFields[fieldName].Unpack(binaryPackageFactory));
```

Reply:
yup, those worked. fixed